airbyte-integrations/connectors/destination-redshift/src/main/resources/spec.json

```diff
@@ -48,76 +48,106 @@
         "default": "public",
         "title": "Default Schema"
       },
-      "s3_bucket_name": {
-        "title": "S3 Bucket Name (Optional)",
-        "type": "string",
-        "description": "The name of the staging S3 bucket to use if utilising a COPY strategy. COPY is recommended for production workloads for better speed and scalability. See <a href=\"https://docs.aws.amazon.com/AmazonS3/latest/userguide/creating-bucket.html\">AWS docs</a> for more details.",
-        "examples": ["airbyte.staging"]
-      },
-      "s3_bucket_path": {
-        "title": "S3 Bucket Path (Optional)",
-        "type": "string",
-        "description": "The directory under the S3 bucket where data will be written. If not provided, then defaults to the root directory. See <a href=\"https://docs.aws.amazon.com/prescriptive-guidance/latest/defining-bucket-names-data-lakes/faq.html#:~:text=be%20globally%20unique.-,For%20S3%20bucket%20paths,-%2C%20you%20can%20use\">path's name recommendations</a> for more details.",
-        "examples": ["data_sync/test"]
-      },
-      "s3_bucket_region": {
-        "title": "S3 Bucket Region (Optional)",
-        "type": "string",
-        "default": "",
-        "description": "The region of the S3 staging bucket to use if utilising a COPY strategy. See <a href=\"https://docs.aws.amazon.com/AmazonS3/latest/userguide/creating-bucket.html#:~:text=In-,Region,-%2C%20choose%20the%20AWS\">AWS docs</a> for details.",
-        "enum": [
-          "",
-          "us-east-1",
-          "us-east-2",
-          "us-west-1",
-          "us-west-2",
-          "af-south-1",
-          "ap-east-1",
-          "ap-south-1",
-          "ap-northeast-1",
-          "ap-northeast-2",
-          "ap-northeast-3",
-          "ap-southeast-1",
-          "ap-southeast-2",
-          "ca-central-1",
-          "cn-north-1",
-          "cn-northwest-1",
-          "eu-central-1",
-          "eu-north-1",
-          "eu-south-1",
-          "eu-west-1",
-          "eu-west-2",
-          "eu-west-3",
-          "sa-east-1",
-          "me-south-1"
+      "uploading_method": {
```

Comment:
I'm a bit concerned with broken backward compatibility. @alexandertsukanov is there a way to pick S3 staging by default and probably read existing config instead of making users update connector settings?
cc @grishick

Reply:
@alexandr-shegeda this was fixed and covered with the unit tests. Great comment. Thanks.

__
[![sema-logo](https://app.semasoftware.com/img/sema-tray-logo.gif)](https://semasoftware.com/gh) &nbsp;**Summary:** :ok_hand: This code looks good